include/mppic/trajectory_visualizer.hpp

```diff
@@ -33,11 +33,11 @@ class TrajectoryVisualizer
     int id, const geometry_msgs::msg::Pose & pose, const geometry_msgs::msg::Vector3 & scale,
     const std_msgs::msg::ColorRGBA & color, const std::string & frame_id);
 
-  geometry_msgs::msg::Pose createPose(double x, double y, double z);
+  geometry_msgs::msg::Pose createPose(const double & x, const double & y, const double & z);
```

Comment:
Well this is where it starts to not matter, doubles are also 8 bytes. You're right on the smaller ones but isn't this probably better to keep with the reference value?

Reply:
reference implicitly is a const pointer which is 8 bytes on x64. You pay here for the pointer and for dereference. 

Core guidelines quote:
[reference](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const) F.16: For “in” parameters, pass cheaply-copied types by value and others by reference to const
Reason Both let the caller know that a function will not modify the argument, and both allow initialization by rvalues.
What is “cheap to copy” depends on the machine architecture, but two or three words (doubles, pointers, references) are usually best passed by value. When copying is cheap, nothing beats the simplicity and safety of copying, and for small objects (up to two or three words) it is also faster than passing by reference because it does not require an extra indirection to access from the function.